client.go

```diff
@@ -858,12 +848,14 @@ func (c *Client[TTx]) StopAndCancel(ctx context.Context) error {
 }
 
 // Stopped returns a channel that will be closed when the Client has stopped.
-// It can be used to wait for a graceful shutdown to complete.
+// It can be used to wait for a graceful shutdown to complete. A reference to
+// the channel must be taken _before_ calling Stop or StopAndCancel. The channel
+// is reset after the service is stopped, so if a reference isn't taken before
+// stopping, then the caller might (it's a race) receive a channel intended for
+// use the next time the client is stopped.
 //
 // It is not affected by any contexts passed to Stop or StopAndCancel.
-func (c *Client[TTx]) Stopped() <-chan struct{} {
-	return c.stopped
-}
+func (c *Client[TTx]) Stopped() <-chan struct{} { return c.baseStartStop.Stopped() }
```

Comment:
So because `BaseStartStop.Stopped` now locks a mutex, I had to change the client here so that instead of take a reference to `Stopped` in `Start` (which no longer works because the `BaseStartStop`'s mutex is held during `Start`), I had to change this so that `Stopped` is a direct passthrough into `BaseStartStop.Stopped`.

This does make the code nicer (no longer necessary to have a local `Client.stopped` variable and initialize it, but does result in a subtle change in behavior which I'll mention in the comment below.

Reply:
(NM, changed this.)